firecloud_project.py

```diff
@@ -353,71 +385,95 @@ def generate_config(context):
   project_name = context.properties.get('projectName', project_id)
   labels_obj = context.properties.get('labels', {})
 
+  # Save this template's version number and all parameters inputs to the project metadata to keep track of what
+  # operations were performed on a project.
+  labels_obj.update({
+    "firecloud-project-template-version" : str(FIRECLOUD_PROJECT_TEMPLATE_VERSION)
+  })
+
+  # Label text requirements include max length of 63 chars, only allowing (a-z, 0-9, -, _), value must start with a
+  # letter, and value must be a string. Link: https://cloud.google.com/deployment-manager/docs/creating-managing-labels#requirements
+  LABEL_MAX_LENGTH = 63
+  ALLOWED_CHARS_COMPLEMENT = r'[^a-z0-9-_]+'
+  ALLOWED_STARTING_CHARS_COMPLEMENT = r'^[^a-z]*'
+  for k, v in context.properties.items():
```

Comment:
Yeah it definitely gets mangled a bit. I posted the output here: https://github.com/broadinstitute/gcp-dm-templates/pull/20#discussion_r399281631

Reply:
I had another thought.. Since the label value is limited to 63 chars, some text gets truncated. We could make multiple labels (labelName, labelName-1, labelName-2, etc.) Would it be worth doing this? I don't think we get much useful information after 63 chars so I'd lean towards no.

Reply:
re: filtering out labels, I wasn't sure which labels would be useful and which wouldn't be, and whether that could change in the future. Having extra labels doesn't feel like a large cost compared to needing to curate the list so I went with leaving them all in. Open to hear differing thoughts though!

Reply:
I say just leave them all in, an example use case I am thinking of is in the future we need to migrate projects with `privateIpGoogleAccess == True`, version is not enough information as some version 2 projects will have `privateIpGoogleAccess == False`.